### PR TITLE
Fix build of paraview

### PR DESCRIPTION
**Pull request by sbna**

### CODE REVIEW

File: var/spack/repos/builtin/packages/mesa/package.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ class Mesa(AutotoolsPackage):
     list_url = "https://mesa.freedesktop.org/archive"
     list_depth = 2
 
-    version('17.1.5', '6cf936fbcaadd98924298a7009e8265d')
+    version('17.1.5', '6cf936fbcaadd98924298a7009e8265d', url='https://mesa.freedesktop.org/archive/mesa-17.1.5.tar.xz')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/mesa/package.py]: Why do u need specify url explicitly?

Comment (sbna) [reply var/spack/repos/builtin/packages/mesa/package.py]: I noticed that the URL of mesa version 17.x is different from the previous versions.
We have for 17.x:
https://mesa.freedesktop.org/archive/mesa-17.x.x.tar.xz

For older versions:
https://mesa.freedesktop.org/archive/older-versions/13.x/13.x.x/mesa-13.x.x.tar.xz

Probably it's better to change the list_url and fix the older urls (for 12.x and 13.x versions)

Comment (davydden) [reply var/spack/repos/builtin/packages/mesa/package.py]: I can confirm that `spack fetch mesa` currently fails with `17.1.5`. Probably this change should rather go into 
```
def url_for_version(self, version):
    if version >= Version('13.0.6'):
      ....
    else:
      ....
```


Then you can conditionally use either old one or new one based on the version.

Comment (sbna) [reply var/spack/repos/builtin/packages/mesa/package.py]: ok

**Comment by davydden** [comment]
while on this branch, you can do `git rebase -i HEAD~2` and then drop the first commit. Then do `git push --force` to force push to your remote (i.e. fork of Spack).

**Pull request merged**